pkg/reconciler/workload/namespace/conditions.go

```diff
@@ -104,3 +104,14 @@ func setScheduledCondition(ns *corev1.Namespace) *corev1.Namespace {
 
 	return updatedNs
 }
+
+// hasWorkloadClusterAssigned returns whether the given namespace has a workload cluster assigned.
+func hasWorkloadClusterAssigned(ns *corev1.Namespace) bool {
+	hasWorkloadAssigned := false
+	for clusterName, val := range workloadClustersFromLabels(ns.Labels) {
+		if val != "" && ns.Annotations[LocationDeletionAnnotationName(clusterName)] == "" {
```

Comment:
hmm, so with this annotation we say there is no cluster assigned anymore? Not sure "assigned" is a good word for that.

Reply:
Is this helper func only for the `Scheduled` condition? Then I think just remove this special case here about the annotation. IMO a namespace being deleted is still assigned and scheduled.